### PR TITLE
deps: bump SDK v0.50.1

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: https://github.com/cosmos/ibc-go/issues/3707

### Commit Message / Changelog Entry

```text
deps: upgrade SDK to v0.50.1
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ go 1.21
 module github.com/cosmos/ibc-go/v8
 
 require (
-	cosmossdk.io/api v0.7.2
+	cosmossdk.io/api v0.7.3-0.20231029200940-6af7f30bfd54
```
Comment (damiannolan) [inline go.mod:6]: any additional info on this?

File: modules/apps/27-interchain-accounts/host/keeper/relay_test.go
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/host/keeper/relay_test.go:1]: curious, why did the values need to be made larger?

**Review by damiannolan — APPROVED** [review]
Looks good to me, thanks @crodriguezvega

File: go.mod (hunk 1 above)
Comment (colin-axner) [inline go.mod:6]: what is this change for? I see v0.50.1, uses v0.7.2 https://github.com/cosmos/cosmos-sdk/blob/v0.50.1/go.mod#L8

Comment (crodriguezvega) [reply go.mod]: I don't know, but I downgraded it v7.2.0 now.

File: modules/apps/callbacks/go.mod [hunk 2]
```diff
@@ -9,7 +9,7 @@ replace github.com/cosmos/ibc-go/v8 => ../../../
 replace github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
 
 require (
-	cosmossdk.io/api v0.7.2
+	cosmossdk.io/api v0.7.3-0.20231029200940-6af7f30bfd54
```
Comment (colin-axner) [inline modules/apps/callbacks/go.mod:12]: same concern here

File: modules/apps/27-interchain-accounts/host/keeper/relay_test.go
Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/host/keeper/relay_test.go:1]: The governance transactions requested to have a minimum deposit of 100000.

File: modules/apps/callbacks/go.mod (hunk 2 above)
Comment (crodriguezvega) [reply modules/apps/callbacks/go.mod:12]: Downgraded as well.

File: go.mod (hunk 1 above)
Comment (crodriguezvega) [reply go.mod]: I downgraded to v0.7.2

File: modules/apps/27-interchain-accounts/host/keeper/relay_test.go
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/host/keeper/relay_test.go:1]: Aha, okay cool!

**Review by damiannolan — APPROVED** [review]
ACK updates 🚀

**Review by colin-axner — APPROVED** [review]
Woot woot!

**Pull request merged**